File: clients/js/src/features/config_env_variables.ts
import { Input } from "../input";

import { PolywrapClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";

export async function runTestCase(input: unknown): Promise<void> {
  const inputObj = Input.expectObject<{
    uri: unknown;
    env: unknown;
  }>(input);
  const uri = Input.expectUri(inputObj.uri).uri;
  const env = Input.expectObject<Record<string, unknown>>(
    inputObj.env
  );

  console.log("Adding Env to ClientConfig");

  const config = new PolywrapClientConfigBuilder()
    .addEnv(uri, env)
    .build();

  const client = new PolywrapClient(config);

  console.log("Fetching Env");

  const result = client.getEnvByUri(uri);

  if (result) {
    for (const key of Object.keys(result)) {
      console.log(`env.${key} = ${result[key]}`);
    }
    console.log("Success!");
  }
}

File: clients/py/src/features/config_env_variables.py
from polywrap_client import PolywrapClient
from polywrap_client_config_builder import PolywrapClientConfigBuilder
from pydantic import BaseModel, validator
from typing import Any, TypedDict
import sys
from validators import UriStr, validate_uri


Env = TypedDict("Env", {"str": str, "num": int})


class TestCaseInput(BaseModel):
    uri: UriStr
    env: Env

    @validator("uri")
    def valid_uri(cls, v: Any):
        return validate_uri(v)


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.validate(input)

    print("Adding Env to ClientConfig")

    config = PolywrapClientConfigBuilder().set_env(input_obj.uri, input_obj.env).build()
    client = PolywrapClient(config)

    print("Fetching Env")

    if result := client.get_env_by_uri(input_obj.uri):
        for key in result.keys():
            print(f"env.{key} = {result[key]}")
        print("Success!")
    else:
        print("Failed to fetch env.", file=sys.stderr)
